### PR TITLE
unit: 1.29.1 -> 1.30.0

### PR DESCRIPTION
**Pull request by Izorkin**
###### Description of changes
Update unit to version 1.30.0.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/http/unit/default.nix [hunk 1]
```diff
@@ -6,9 +6,9 @@
 , withPerl534 ? false, perl534
 , withPerl536 ? true, perl536
 , withPerldevel ? false, perldevel
-, withRuby_2_7 ? true, ruby_2_7
 , withRuby_3_0 ? false, ruby_3_0
-, withRuby_3_1 ? false, ruby_3_1
+, withRuby_3_1 ? true, ruby_3_1
+, withRuby_3_1 ? false, ruby_3_2
```
Comment (JulienMalka) [inline pkgs/servers/http/unit/default.nix:11]:
```suggestion
, withRuby_3_2? false, ruby_3_2
```

Comment (Izorkin) [reply pkgs/servers/http/unit/default.nix]: Thanks, fixed.

**Review by ajs124 — CHANGES_REQUESTED** [review]
fails to build for me:
```
        /nix/store/skr7af4dd35ysk95pic756nc3s6kmazw-unit-1.30.0/lib/unit/modules/
install -d /nix/store/skr7af4dd35ysk95pic756nc3s6kmazw-unit-1.30.0/lib/unit/modules
install -p build/lib/unit/modules/php81.unit.so \
        /nix/store/skr7af4dd35ysk95pic756nc3s6kmazw-unit-1.30.0/lib/unit/modules/
install -d /nix/store/skr7af4dd35ysk95pic756nc3s6kmazw-unit-1.30.0/lib/unit/modules
install -p build/lib/unit/modules/perl536.unit.so \
        /nix/store/skr7af4dd35ysk95pic756nc3s6kmazw-unit-1.30.0/lib/unit/modules/
install -d /nix/store/skr7af4dd35ysk95pic756nc3s6kmazw-unit-1.30.0/lib/unit/modules
install -p build/lib/unit/modules/ruby31.unit.so \
        /nix/store/skr7af4dd35ysk95pic756nc3s6kmazw-unit-1.30.0/lib/unit/modules/
rmdir: failed to remove '/nix/store/skr7af4dd35ysk95pic756nc3s6kmazw-unit-1.30.0/state': No such file or directory
/nix/store/smcpmkfc7bx90ry7wmr6nckg83anjj0k-stdenv-linux/setup: line 136: pop_var_context: head of shell_variables not a function context
```

**Comment by Izorkin** [comment]
Fixed. Sorry, I was in a hurry to do PR and didn't fully check.

**Review by ajs124 — APPROVED** [review]

**Pull request merged**